### PR TITLE
make loss cond. on label while using Stirling approximation

### PR DESCRIPTION
**Pull request by akshatvishu**
### PR types
 Others 

### PR changes
 Others 

### Description
This commit updates the condition for adding the Stirling approximation term in Poisson NLL Loss to align with how PyTorch and TensorFlow handle it. The condition is changed from 'label <= 1' to 'label > 1' for adding the Stirling term, making the behavior consistent across frameworks.

### CODE REVIEW

**Comment by akshatvishu** [comment]
Close #56824 #56733

**Comment by luotao1** [comment]
@LyndonKong Could you help review this PR?

**Comment by LyndonKong** [comment]
LGTM

**Review by luotao1 — APPROVED** [review]

**Comment by akshatvishu** [comment]
Thank you @luotao1  and @LyndonKong  for sticking me with this one even though I made a lot of mistakes . Your support and guidance was invaluable !
